integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonDataMapShowCommand.scala

```diff
@@ -92,7 +100,44 @@ case class CarbonDataMapShowCommand(tableIdentifier: Option[TableIdentifier])
               .map(p => s"'${ p._1 }'='${ p._2 }'").toSeq
               .sorted.mkString(", ")
           }
-        Row(s.getDataMapName, s.getProviderName, table, dmPropertieStr)
+          // Get datamap status and sync information details
+          var dataMapStatus = "NA"
+          var syncInfo: String = "NA"
+          if (!s.getProviderName.equalsIgnoreCase(
+            DataMapClassProvider.PREAGGREGATE.getShortName) && !s.getProviderName.equalsIgnoreCase(
+            DataMapClassProvider.TIMESERIES.getShortName)) {
+            if (DataMapStatusManager.getEnabledDataMapStatusDetails
+              .exists(_.getDataMapName.equalsIgnoreCase(s.getDataMapName))) {
+              dataMapStatus = DataMapStatus.ENABLED.name()
+            } else {
+              dataMapStatus = DataMapStatus.DISABLED.name()
+            }
+            val loadMetadataDetails = SegmentStatusManager
+              .readLoadMetadata(CarbonTablePath
+                .getMetadataPath(s.getRelationIdentifier.getTablePath))
+            if (!s.isIndexDataMap && loadMetadataDetails.nonEmpty) {
+              breakable({
+                for (i <- loadMetadataDetails.length - 1 to 0 by -1) {
+                  if (loadMetadataDetails(i).getSegmentStatus.equals(SegmentStatus.SUCCESS)) {
+                    val segmentMaps =
+                      DataMapSegmentStatusUtil.getSegmentMap(loadMetadataDetails(i).getExtraInfo)
+                    val syncInfoMap = new util.HashMap[String, String]()
+                    val iterator = segmentMaps.entrySet().iterator()
+                    while (iterator.hasNext) {
+                      val entry = iterator.next()
+                      syncInfoMap.put(entry.getKey, entry.getValue.get(entry.getValue.size() - 1))
+                    }
+                    syncInfoMap
+                      .put(CarbonCommonConstants.LOAD_SYNC_TIME,
+                        loadMetadataDetails(i).getLoadEndTime.toString)
```

Comment:
i suggest to get the formatted timestamp, like we do in show segments case

Reply:
done